assets/js/keys.js

```diff
@@ -1,56 +1,128 @@
+var searchTerm = ''
+var searchResults = [];
+
+//get the product url for the first 3 results from Amazon
+function getAmazonUrl() {
+	fetch(`https://axesso-axesso-amazon-data-service-v1.p.rapidapi.com/amz/amazon-search-by-keyword-asin?sortBy=relevanceblender&domainCode=com&keyword=${searchTerm}&page=1`, {
+		"method": "GET",
+		"headers": {
+			"x-rapidapi-host": amazonHost,
+			"x-rapidapi-key": axsessoKey
+		}
+	})
+	.then(response => {
+		if (response.ok) {
+			response.json().then(function(data){
+				//loop through the top three results to get asin
+				for(i =0; i < 3; i++) {
+					var asin = data.searchProductDetails[i].asin;
+					//load asin to get specific product details
+					getAmazonProduct(asin);
+				}
+			});
+		}
+	});
+};
+
+
+//get the product url for the first 3 results from Walmart
+function getWalmartUrl() {
+	fetch(`https://axesso-walmart-data-service.p.rapidapi.com/wlm/walmart-search-by-keyword?sortBy=best_match&page=1&keyword=${searchTerm}&type=text`, {
+		"method": "GET",
+		"headers": {
+			"x-rapidapi-host": walmartHost,
+			"x-rapidapi-key": axsessoKey
+		}
+	})
+	.then(response => {
+		if (response.ok) {
+			response.json().then(function(data) {
+				for (i=0; i < 3; i++) {
+					var produrl = data.foundProducts[i];
+					//load product url to get specific product details
+					getWalmartProduct(produrl);
+				}
+			});
+		}
+	});
+};
+
+function getWalmartProduct(produrl) {
+	fetch("https://axesso-walmart-data-service.p.rapidapi.com/wlm/walmart-lookup-product?url=https://www.walmart.com" + produrl, {
+		"method": "GET",
+		"headers": {
+			"x-rapidapi-host": "axesso-walmart-data-service.p.rapidapi.com",
+			"x-rapidapi-key": "LIf3v3u97Wmshek4PIKJGfwmDRHHp1e33VnjsnxVU7ZUW0fu5W"
+		}
+	})
+	.then(response => {
+		if (response.ok) {
+			response.json().then(function(data){
+				var productDetails = makeWalmartProduct(data);
+				searchResults.push(productDetails);
+			});
+		}
+	});
+};
+
+
+function getAmazonProduct(asin) {
+	fetch("https://axesso-axesso-amazon-data-service-v1.p.rapidapi.com/amz/amazon-lookup-product?url=https://www.amazon.com/dp/" + asin, {
+		"method": "GET",
+		"headers": {
+			"x-rapidapi-host": "axesso-axesso-amazon-data-service-v1.p.rapidapi.com",
+			"x-rapidapi-key": "LIf3v3u97Wmshek4PIKJGfwmDRHHp1e33VnjsnxVU7ZUW0fu5W"
+		}
+	})
+	.then(response => {
+		//verify a result was received
+		if (response.ok) {
+			response.json().then(function(data){
+				var productDetails = makeAmazonProduct(data);
+				searchResults.push(productDetails);
+			})
+		}
+	});
+};
+
+
+function makeAmazonProduct(product) {
+	return {
+		title: product.productTitle,
+		price: product.price || '',
+		availability: isAvailable(product.warehouseAvailability),
+		prime: product.prime,
+		url: `https://www.amazon.com/dp/${product.asin}`
+	};
+}
+
+function makeWalmartProduct(product) {
+	return {
+		title: product.productTitle,
+		price: product.price || '',
+		availability: product.available,
```

Comment:
Do we need to call `isAvailable()` here too?

Reply:
Fortunately, no. Walmart has a boolean property 'available' on the product response so we can just save it to the product object.